src/Tests/Microsoft.TemplateEngine.Cli.UnitTests/AliasAssignmentTests.cs

```diff
@@ -200,5 +207,272 @@ public void CheckAliasAssignmentsMvc20()
+        [MemberData(nameof(GetTemplateData))]
+#pragma warning restore CA1825 // Avoid zero-length array allocations.
+        public void CanOverrideAliasesForParameterWithHostData(string hostJsonData, string expectedJsonResult)
+        {
+            var hostData = new HostSpecificTemplateData(string.IsNullOrEmpty(hostJsonData) ? null : JObject.Parse(hostJsonData));
+            var expectedResults = JObject.Parse(expectedJsonResult);
+            var template = new MockTemplateInfo("foo", identity: "foo.1", groupIdentity: "foo.group");
+            foreach (var expectedResult in expectedResults)
+            {
+                template.WithParameter(expectedResult.Key);
+            }
+            var hostDataLoader = A.Fake<IHostSpecificDataLoader>();
+            A.CallTo(() => hostDataLoader.ReadHostSpecificTemplateData(template)).Returns(hostData);
+            TemplateGroup templateGroup = TemplateGroup.FromTemplateList(
+                CliTemplateInfo.FromTemplateInfo(new[] { template }, hostDataLoader))
+                .Single();
+            ICliTemplateEngineHost host = CliTestHostFactory.GetVirtualHost();
+            IEngineEnvironmentSettings settings = new EngineEnvironmentSettings(host, virtualizeSettings: true);
+            TemplatePackageManager templatePackageManager = A.Fake<TemplatePackageManager>();
+            NewCommand myCommand = (NewCommand)NewCommandFactory.Create("new", _ => host);
+            ParseResult parseResult = myCommand.Parse(" new foo");
+            InstantiateCommandArgs args = InstantiateCommandArgs.FromNewCommandArgs(new NewCommandArgs(myCommand, parseResult));
+            var templateCommands = InstantiateCommand.GetTemplateCommand(args, settings, templatePackageManager, templateGroup);
+            Assert.Single(templateCommands);
+            foreach (var expectedResult in expectedResults)
+            {
+                var expectedValues = expectedResult.Value!.Select(s => ((JValue)s).Value).ToArray();
+                var expectedLongAlias = expectedValues[0];
+                var expectedShortAlias = expectedValues[1];
+                var expectedIsHidden = expectedValues[2];
+                var templateOptions = templateCommands.Single().TemplateOptions;
+                Assert.NotNull(templateOptions);
+                Assert.Contains(expectedResult.Key, templateOptions.Keys);
+                var templateOption = templateOptions[expectedResult.Key];
+                Assert.NotNull(templateOption);
+                Assert.True(templateOption.Aliases.Count > 0);
+                //var longAlias = templateOption.Aliases[0];
+                //var shortAlias = templateOption.Aliases.Count > 1 ? templateOption.Aliases[1] : null;
+                //var isHidden = templateOption.Option.IsHidden;
+                //Assert.Equal(expectedLongAlias, longAlias);
+                //Assert.Equal(expectedShortAlias, shortAlias);
+                //Assert.Equal(expectedIsHidden, isHidden);
```

Comment:
please uncomment or remove if not needed

Reply:
my fault, I forgot to uncomment it after some debugging.